planetscale/workflows.go

```diff
@@ -0,0 +1,185 @@
+}
+
+type WorkflowVDiffTableReport struct {
+	PublicID                   string        `json:"id"`
+	TableName                  string        `json:"table_name"`
+	Shard                      string        `json:"shard"`
+	MismatchedRowsCount        int64         `json:"mismatched_rows_count"`
+	ExtraSourceRowsCount       int64         `json:"extra_source_rows_count"`
+	ExtraTargetRowsCount       int64         `json:"extra_target_rows_count"`
+	ExtraSourceRows            []interface{} `json:"extra_source_rows"`
+	ExtraTargetRows            []interface{} `json:"extra_target_rows"`
+	MismatchedRows             []interface{} `json:"mismatched_rows"`
+	SampleExtraSourceRowsQuery string        `json:"sample_extra_source_rows_query"`
+	SampleExtraTargetRowsQuery string        `json:"sample_extra_target_rows_query"`
+	SampleMismatchedRowsQuery  string        `json:"sample_mismatched_rows_query"`
+	CreatedAt                  time.Time     `json:"created_at"`
+	UpdatedAt                  time.Time     `json:"updated_at"`
+}
+
+type ListWorkflowsRequest struct {
+	Organization string `json:"-"`
+	Database     string `json:"-"`
+}
+
+type GetWorkflowRequest struct {
+	Organization   string `json:"-"`
+	Database       string `json:"-"`
+	WorkflowNumber int    `json:"-"`
+}
+
+// WorkflowsService is an interface for interacting with the keyspace endpoints of the PlanetScale API
+type WorkflowsService interface {
+	List(context.Context, *ListWorkflowsRequest) ([]*Workflow, error)
+	Get(context.Context, *GetWorkflowRequest) (*Workflow, error)
+}
+
+type workflowsService struct {
+	client *Client
+}
+
+var _ WorkflowsService = &workflowsService{}
+
+func NeWorkflowsService(client *Client) *workflowsService {
+	return &workflowsService{client}
+}
+
+func (ws *workflowsService) List(ctx context.Context, listReq *ListWorkflowsRequest) ([]*Workflow, error) {
+	req, err := ws.client.newRequest(http.MethodGet, workflowsAPIPath(listReq.Organization, listReq.Database), nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	workflows := &workflowsResponse{}
+
+	if err := ws.client.do(ctx, req, workflows); err != nil {
+		return nil, err
+	}
+
+	return workflows.Workflows, nil
+}
+
+func (ws *workflowsService) Get(ctx context.Context, getReq *GetWorkflowRequest) (*Workflow, error) {
+	req, err := ws.client.newRequest(http.MethodGet, workflowAPIPath(getReq.Organization, getReq.Database, getReq.WorkflowNumber), nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	workflow := &Workflow{}
+
+	if err := ws.client.do(ctx, req, workflow); err != nil {
+		return nil, err
+	}
+
+	return workflow, nil
+}
+
+func workflowsAPIPath(org, db string) string {
+	return fmt.Sprintf("%s/%s/workflows", databasesAPIPath(org), db)
+}
+
+func workflowAPIPath(org, db string, number int) string {
+	return fmt.Sprintf("%s/%d", workflowsAPIPath(org, db), number)
+}
+
+type workflowsResponse struct {
```

Comment:
Might be worth moving this closer to where it's used in `List`

Reply:
Done ✨